src/Connection/StreamConnection.php

```diff
@@ -316,9 +316,20 @@ private function handleError(ErrorResponseInterface $error, CommandInterface $fa
             if (in_array('AUTH', $failedCommand->getArguments(), true)) {
                 $parameters = $this->getParameters();
 
+                // If Redis > 6.2
                 $auth = new RawCommand('AUTH', [$parameters->username, $parameters->password]);
```

Comment:
If I understand correctly, this would run after we've gotten an error on the AUTH we sent in the factory. If that's the case, why repeat the same command (with username and pass), why not go straight to the pre 6.0 version with no username?

Reply:
Yeah, I think you're right, before 6.0 username wasn't an option so there's no way there's case when before 6.0 when AUTH wouldn't work without username

Reply:
Done